### ISSUE TITLE
Did u try the e2c model on the simulated inverted pendulum test set ?

### ISSUE THREAD

**Issue by ZhengYi0310** [pclucas14/e2c#1]
Hi,
   My name is Yi Zheng, currently I'm also implementing the e2c model [my implementation](https://github.com/ZhengYi0310/other_stuff/blob/gh-pages/deeprlhw/hw1/e2c.py). But when I tried my implementation on the simulated inverted pendumlum dataset, I got the following result:
![inverted_pendulum_result](https://user-images.githubusercontent.com/8685125/32202200-548480c6-bd98-11e7-8e13-66e0f58cc160.png)

pretty much it failed completely, just to double check if there is anything wrong in my implementation, I git clone your implementation and tweak the datasets.py and train_e2c.py files to test it on the simulated inverted pendulum dataset, you can see the tweak from the commit history, it still completely failed:
![screenshot from 2017-10-30 17 56 05](https://user-images.githubusercontent.com/8685125/32203042-18423ea4-bd9e-11e7-95da-b231d4498aa1.png)

So just want to know did you every tested it on the simulated inverted pendulum dataset ? And I saw you using it on your lidar data, how does it work on the real data ?

Thanks very much for your time and patience !

**Comment by ZhengYi0310** [comment]
Looks like I scale the images at the wrong place, before when I load the dataset in datasets.py, I divide the image array by 255. to convert it to 0~1.0. Now instead of doing this, I scale the decoder output by 255. and it gives me the following, though I don't quite understand why this would happen.
![screenshot from 2017-10-30 20 23 36](https://user-images.githubusercontent.com/8685125/32206072-43202d86-bdb0-11e7-814d-e907d03ffb95.png)

**Comment by ZhengYi0310** [comment]
Hi, 
     I still find some problems, did u try to visualize your latent space after training, this is what looks like after I trained the encoder and get the reconstruction:
![screenshot from 2017-10-31 20 46 35](https://user-images.githubusercontent.com/8685125/32259288-a2d6dfa0-be7c-11e7-8ea7-dbeeb5d81d64.png)

it looks like although the model learned the reconstruction, but it failed to learn the latent embedding, I notice that when you compute reconstruction loss you are using mse error rather than the cross entropy, wouldn't that make the model AE rather than VAE ?

If I change the reconstruction loss back to cross entropy, then the learning process will diverge and return nan cost.

Sorry to bug you like this, but this is really confusing, do you have any idea or can you reproduce the result from the vae paper on using your own implementation and datasets?

Thanks very much for your time and patience!
